nomad-test/src/test_utils.rs

```diff
@@ -68,6 +69,28 @@ where
     assert!(result.is_ok())
 }
 
+/// Run test with a mock http server response and env
+pub async fn run_test_with_env_http<T, Fut>(path: impl AsRef<Path>, body: impl AsRef<[u8]>, test: T)
```

Comment:
Hard to intuit what this hook does given name. Maybe `run_test_serve_remote_config` and it only runs the server (does nothing with env vars). Then we can call the two hooks nested instead like:
```
...
   test_utils::run_test_with_env(
      "../../fixtures/env.external",
      test_utils::run_test_serve_remote_config(
         json_config,
          ... test logic ...
      ),
   )
...
```

Reply:
I changed it to `run_test_with_http_response` but kept it general enough that it will accept / return an response body. We may want it to test other things in the future.

Didn't think about nesting them for composability, good idea.